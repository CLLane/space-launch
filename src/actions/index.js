export const nextLaunchData = (launch) => ({
  type: 'NEXT_LAUNCH',
  launch,
})

export const monthLaunchData = (launches) => ({
  type: 'MONTH_LAUNCH',
  launches
})

export const rocketTypeData = (rockets) => ({
  type: 'ROCKET_TYPE',
  rockets
})

export const getMissionData = (missions) => ({
  type: 'MISSION_DATA',
  missions
})

export const getCompanyData = (companies) => ({
  type: 'COMPANY_DATA',
  companies
})