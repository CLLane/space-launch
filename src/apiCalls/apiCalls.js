export const getLaunchSchedule = async (number) => {
  try {
    const response = await fetch(`https://launchlibrary.net/1.3/launch/next/${number}`)
    const data = await response.json();
    return data.launches;
  } catch (error) {
    console.log(error.message)
    throw new Error('Failed to retrieve scheduled launches')
  }
}

export const monthLaunchSchedule = async (start, end) => {
  try {
    const response = await fetch(`https://launchlibrary.net/1.3/launch/${start}/${end}`);
    const data = await response.json();
    return data.launches
  } catch (error) {
    console.log(error.message)
    throw new Error('Failed to retrieve the months scheduled launches')
  }
}

export const getRocketTypes = async () => {
  try {
    const response = await fetch('https://launchlibrary.net/1.3/rocketfamily/');
    const data = await response.json();
    return data.RocketFamilies
  } catch (error) {
    throw new Error('Failed to retrieve Rocket Types')
  }
}

export const getCompanies = async () => {
  try {
    const response = await fetch('https://launchlibrary.net/1.3/agency');
    const data = await response.json();
    return data.agencies
  } catch (error) {
    throw new Error('Failed to retrieve Companies')
  }
}