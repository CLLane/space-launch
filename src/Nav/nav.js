import React from "react";
import { Component } from "react";
import { getSearchResults } from "../apiCalls/apiCalls";
import { getSearchData } from "../actions/index";
import { connect } from "react-redux";
import { NavLink } from "react-router-dom";
import Logo from '../Images/logo.svg'

export class Nav extends Component {
  constructor() {
    super();
    this.state = {
      search: "",
      condition: ""
    };
  }

  handleChange = e => {
    this.setState({ search: e.target.value });
  };

  handleCondition = e => {
    this.setState({ condition: e.target.value });
  };

  fetchSearchResults = async (condition, search) => {
    let searchFetch = await getSearchResults(condition, search);
    await this.props.getSearchData(searchFetch);
  };

  render() {
    return (
      <>
      <div className='image-container'>
       <img className='logo' alt='' src={Logo}></img>
      <p>Launch Log</p>
      </div>
      <form>

        <div className='radio-group'>
          <div className='radio'>
            <input
              type="radio"
              name="condition"
              value="launch/"
              onClick={this.handleCondition}>
            </input>
            Launch</div>
          <div className='radio'>
            <input
              className='radio'
              type="radio"
              name="condition"
              value="agency?name="
              onClick={this.handleCondition}>
            </input>
            Company</div>
          <div className='radio'>
            <input
              className='radio'
              type="radio"
              name="condition"
              value="rocket/"
              onClick={this.handleCondition}>
            </input>
            Rocket</div>
          <div className='radio'>
            <input
              className='radio'
              type="radio"
              name="condition"
              value="mission?name="
              onClick={this.handleCondition}>
            </input>
            Missions</div>
        </div>
        <div className='search-container'>
          <input
            type="text"
            value={this.state.search}
            onChange={this.handleChange}
          ></input>
          {!this.state.search && !this.state.condition && (
            <button disabled>Search</button>
          )}
          {this.state.search && !this.state.condition && (
            <button disabled>Search</button>
          )}
          {!this.state.search && this.state.condition && (
            <button disabled>Search</button>
          )}
          {this.state.search && this.state.condition && (
            <NavLink className='active-button' to="/Search"
                onClick={() =>
                  this.fetchSearchResults(this.state.condition, this.state.search)
                }
              >
                Search
            </NavLink>
          )}
        </div>
      </form>
      </>
    );
  }
}

export const mapDispatchToProps = dispatch => ({
  getSearchData: search => dispatch(getSearchData(search))
});

export const mapStateToProps = ({ search }) => ({
  search
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Nav);
