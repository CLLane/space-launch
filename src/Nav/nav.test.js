import React from 'react';
import { shallow } from 'enzyme';
import { Nav, mapStateToProps, mapDispatchToProps } from "./nav";
import { getSearchData } from "../actions/index";
import { getSearchResults } from '../apiCalls/apiCalls';


describe('Nav', () => {
  let wrapper;
  const mockGetSearchData = jest.fn();
  const mockHandleCondition = jest.fn();
  const mockHandleChange = jest.fn();
  const mockSearch = [{name: 'result'}];

  beforeEach(() => {
    wrapper = shallow(
      <Nav 
        search={mockSearch}
        getSearchData={mockGetSearchData}
      />
    )
  })
  it.only('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})