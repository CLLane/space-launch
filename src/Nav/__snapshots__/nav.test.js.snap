// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should match snapshot 1`] = `
<Fragment>
  <div
    className="image-container"
  >
    <img
      alt=""
      className="logo"
      src="logo.svg"
    />
    <p>
      Launch Log
    </p>
  </div>
  <form>
    <div
      className="radio-group"
    >
      <div
        className="radio"
      >
        <input
          name="condition"
          onClick={[Function]}
          type="radio"
          value="launch/"
        />
        Launch
      </div>
      <div
        className="radio"
      >
        <input
          className="radio"
          name="condition"
          onClick={[Function]}
          type="radio"
          value="agency?name="
        />
        Company
      </div>
      <div
        className="radio"
      >
        <input
          className="radio"
          name="condition"
          onClick={[Function]}
          type="radio"
          value="rocket/"
        />
        Rocket
      </div>
      <div
        className="radio"
      >
        <input
          className="radio"
          name="condition"
          onClick={[Function]}
          type="radio"
          value="mission?name="
        />
        Missions
      </div>
    </div>
    <div
      className="search-container"
    >
      <input
        onChange={[Function]}
        type="text"
        value=""
      />
      <button
        disabled={true}
      >
        Search
      </button>
    </div>
  </form>
</Fragment>
`;
